### PR TITLE
Make some text selectable and add mouse cursors to clickable things

### PR DESCRIPTION
**Pull request by guidezpl**
Did a pass-through, replacing `Text` with `SelectableText` for text which wasn't already clickable.

Also added click mouse cursors to clickable things that were lacking it (`GestureDetector`).

Fixes https://github.com/flutter/gallery/issues/499
Fixes https://github.com/flutter/gallery/issues/529

### CODE REVIEW

File: lib/studies/shrine/expanding_bottom_sheet.dart [hunk 1]
```diff
@@ -741,6 +744,8 @@ class ProductThumbnail extends StatelessWidget {
   Widget build(BuildContext context) {
     final isDesktop = isDisplayDesktop(context);
 
+    print(opacityAnimation);
+    print(animation);
```
Comment (rami-a) [inline lib/studies/shrine/expanding_bottom_sheet.dart:748]: Remove these print statements

File: lib/studies/shrine/login.dart [hunk 2]
```diff
@@ -59,7 +59,6 @@ class LoginPage extends StatelessWidget {
               ),
             )
           : Scaffold(
-              appBar: AppBar(backgroundColor: Colors.white),
```
Comment (rami-a) [inline lib/studies/shrine/login.dart:62]: Why was this removed?

File: lib/studies/rally/login.dart [hunk 3]
```diff
@@ -39,7 +39,6 @@ class _LoginPageState extends State<LoginPage> with RestorationMixin {
   Widget build(BuildContext context) {
     return ApplyTextOptions(
       child: Scaffold(
-        appBar: AppBar(automaticallyImplyLeading: false),
```
[the remark this answers is not in the document]
Comment (guidezpl) [reply lib/studies/rally/login.dart:42]: No need for it, and it was showing up as the default blue

File: lib/studies/shrine/login.dart (hunk 2 above)
Comment (guidezpl) [reply lib/studies/shrine/login.dart:62]: Same as for Rally, no need for it (and there's already the back button)

File: lib/pages/home.dart [hunk 4]
```diff
@@ -564,13 +564,11 @@ class _DesktopCategoryHeader extends StatelessWidget {
...
               child: Semantics(
                 header: true,
-                child: Text(
+                child: SelectableText(
                   category.displayTitle(GalleryLocalizations.of(context)),
                   style: Theme.of(context).textTheme.headline5.apply(
                         color: colorScheme.onSurface,
                       ),
-                  maxLines: 4,
-                  overflow: TextOverflow.ellipsis,
```
[the remark this answers is not in the document]
Comment (guidezpl) [reply lib/pages/home.dart:573]: There's no overflow for SelectableText, and they weren't needed, even at large text scales
